src/components/AttachmentModal.js

```diff
@@ -44,28 +56,78 @@ const propTypes = {
 const defaultProps = {
     title: '',
     sourceURL: null,
+    file: {name: null},
     onConfirm: null,
-    isAuthTokenRequired: false,
+    currentAction: {
+        actionName: '', person: [], sequenceNumber: 0, timestamp: 0, message: []
+    },
+    children: null,
+    setAttachmentModalData: () => { },
+    getNextAttachment: () => { }
 };
 
 class AttachmentModal extends Component {
     constructor(props) {
         super(props);
-
         this.state = {
             isModalOpen: false,
             file: null,
             sourceURL: props.sourceURL,
         };
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.sourceURL !== this.props.sourceURL) {
```

Comment:
I think there is a bug here. What if you close the attachment modal and click it again. Because of this condition won't the attachment modal not show up again?